### PR TITLE
Feature/az mon unified diag settings2

### PR DESCRIPTION
**Pull request by bdschaap**
Configures Service Bus diagnostic settings

### CODE REVIEW

File: ingredient/ingredient-event-hub-namespace/src/index.ts [hunk 1]
```diff
@@ -8,3 +8,5 @@ exports.pluginNS = "@azbake/ingredient-event-hub-namespace" //name of the ingred
 /* comment out these entries if you are not including a set of expression functions*/
 exports.functions = EventHubNamespaceUtils
 exports.functionsNS = "eventhubnamespace" //bake.yaml expressions can access your functions via "myutils.my_function()"
+
+export {EventHubNamespaceUtils}
```
Comment (whilke) [inline ingredient/ingredient-event-hub-namespace/src/index.ts]: don't do this. Will give odd results with how bake uses ingredients

File: ingredient/ingredient-service-bus-namespace/package.json [hunk 2]
```diff
@@ -35,7 +35,8 @@
   },
   "dependencies": {
     "@azbake/arm-helper": "^0.1.55",
-    "@azure/arm-servicebus": "^3.2.0"
+    "@azure/arm-servicebus": "^3.2.0",
+    "@azbake/ingredient-event-hub-namespace": "^0.1.55"
```
Comment (whilke) [inline ingredient/ingredient-service-bus-namespace/package.json]: shouldn't include one ingredient as a dep for another. Use the system for getting a util function handler, with instructions that this ingredient requires including the other ingredient.

File: ingredient/ingredient-service-bus-namespace/src/plugin.ts [hunk 3]
```diff
@@ -1,6 +1,8 @@
 import { BaseIngredient, IngredientManager } from "@azbake/core"
 import { ARMHelper } from "@azbake/arm-helper"
 import ARMTemplate from "./arm.json"
+import { EventHubNamespaceUtils } from "@azbake/ingredient-event-hub-namespace"
```
Comment (whilke) [inline ingredient/ingredient-service-bus-namespace/src/plugin.ts]: don't do this, use the system for loading util handlers

Comment (bdschaap) [reply ingredient/ingredient-service-bus-namespace/src/plugin.ts]: Can you clarify what you mean by using the system for loading util handlers?  

One alternative to my current approach is to push the logic out to the recipe.  I used my current approach for enabling the diagnostic settings on the Event Hub Namespace itself (to transmit logs to a special diagnostics event hub).  However, I didn't want it complicate the recipes so I put it in the deployment logic.

File: ingredient/ingredient-service-bus-namespace/package.json (hunk 2 above)
Comment (bdschaap) [reply ingredient/ingredient-service-bus-namespace/package.json]: See other comment

File: ingredient/ingredient-event-hub-namespace/src/index.ts (hunk 1 above)
Comment (bdschaap) [reply ingredient/ingredient-event-hub-namespace/src/index.ts]: See other comment

File: ingredient/ingredient-service-bus-namespace/src/plugin.ts (hunk 3 above)
Comment (bdschaap) [reply ingredient/ingredient-service-bus-namespace/src/plugin.ts]: It dawned on me that you mean loading utils for another ingredient through IngredientManager.getIngredientFunction.  Similar to how we're loading coreutils in various ingredients.  I'll make this change.

I also reviewed my code again and realize an issue with how I'm pulling the resource group of a different ingredient.  That's what I get for testing with rgOverride.... :-)  I previously solved this in https://github.com/HomecareHomebase/azure-bake/pull/52 by modifying ingredient-utils resource_group() to accept additional params.  I can make the same change in this PR.  Or if the other PR can be reviewed again then I'll merge it into this branch?

Comment (whilke) [reply ingredient/ingredient-service-bus-namespace/src/plugin.ts]: Add that change here so we can push this through and then look at the other PR

**Comment by whilke** [comment]
This is good now. Let's get #87 into master so that you can pull and merge so there isn't a conflict with your same changes.

**Comment by bdschaap** [comment]
@whilke I updated #87  if that can be merged to master

**Comment by whilke** [comment]
@bdschaap can you fix the conflict on this real quick?

**Pull request merged**